canvas.h

```diff
@@ -138,38 +139,10 @@ inline void from_json(const nlohmann::json& j, unique_ptr<ICanvas>& canvas)
         j.value("fps", 30u) 
     );
 
-    // Deserialize features if present
-    if (j.contains("features")) 
-    {
-        for (const auto& featureJson : j["features"])
-            canvas->AddFeature(featureJson.get<unique_ptr<ILEDFeature>>());
-    }
+    // Features()
+    for (const auto& featureJson : j.value("features", nlohmann::json::array()))
+        canvas->AddFeature(featureJson.get<unique_ptr<ILEDFeature>>());
 
-    // Deserialize the EffectsManager
-    if (j.contains("effectsManager"))
-    {
-        auto & effectsManager = canvas->Effects();  // Get reference to existing manager
-        const auto & managerJson = j["effectsManager"];
-
-        // Set FPS if present
-        if (managerJson.contains("fps")) 
-            effectsManager.SetFPS(managerJson["fps"].get<uint32_t>());    
-
-        // Load effects
-        if (managerJson.contains("effects")) 
-        {
-            for (const auto& effectJson : managerJson["effects"]) 
-            {
-                unique_ptr<ILEDEffect> effect;
-                from_json(effectJson, effect);
-                if (effect) 
-                    effectsManager.AddEffect(std::move(effect));
-                
-            }
-        }
-
-        // Set current effect index
-        if (managerJson.contains("currentEffectIndex")) 
-            effectsManager.SetCurrentEffectIndex(managerJson["currentEffectIndex"].get<size_t>());
-    }
-}
\ No newline at end of file
+    // Effects()
+    from_json(j.at("effectsManager"), canvas->Effects());
```

Comment:
Ha, I clearly didn't catch the discrepancy between the old to_json and from_json code... 

As a brevity fan :), I certainly don't mind how much shorter this code has become!

Reply:
I've just extracted some other commanality into FeatureSet() and then call it in the const and non-const case, so it's only one impl.  Will check that in now :-)